lib/react.dart

```diff
@@ -5,14 +5,28 @@
 /// A Dart library for building UI using ReactJS.
 library react;
 
+import 'dart:html' show Element;
+
 /// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/top-level-api.html#react.component)
 /// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/component-api.html)
 abstract class Component {
+  Map _props;
+  Map _state = {};
+  dynamic _ref;
+
   /// ReactJS `Component` props.
-  Map props;
+  Map get props => _props;
+  set props(Map value) => _props = value;
```

Comment:
Actually just tried this, still does not work. I found this issue https://github.com/dart-lang/sdk/issues/27452, which makes it seem like strong mode does not allow you to override fields, even with the `@virtual` annotation. I think that may just be for overriding fields with other fields.